Add scannable option for Domain Text fields

Sigmonia opened the pull request:
#### Rationale
spec: https://docs.google.com/document/d/1wKe6bg2vnJmoa7DA1r43QeO-rt1Wd-2w3yIZen30Neg/edit#

#### Related Pull Requests
* Platform: https://github.com/LabKey/platform/pull/2845
* Biologics: https://github.com/LabKey/biologics/pull/1086
* Inventory: https://github.com/LabKey/inventory/pull/341
* SM: https://github.com/LabKey/sampleManagement/pull/771

#### Changes
* Item #9633: Add scannable option to SampleType text fields for LabKey apps
  * Added scannable field and value property to `TextFieldOptions`
  * Passed through appPropertiesOnly to `DomainRowExpandedOptions` and `TextFieldOptions`
  * Added optional showScannableOption property to `DomainFormDisplayOptions`
  * Added showScannableOption property to `SampleTypeDesigner`
  * Added `scannable` property to `DomainField`

Sigmonia committed:
Adding Domain Designer option to include text field in barcode search query.

Sigmonia committed:
adjust help tip text
Export scannable property

Sigmonia committed:
v2.102.0-fb-scanOption.0

Sigmonia committed:
draft release notes

Sigmonia committed:
update test Snapshots with new properties

Sigmonia committed:
Update test with new field

Sigmonia committed:
adjust TextFieldOptions tests for new control

cnathe commented on packages/components/src/internal/components/domainproperties/TextFieldOptions.tsx:
```diff
@@ -8,7 +8,7 @@ import { isFieldFullyLocked } from './propertiesUtil';
     DOMAIN_FIELD_CUSTOM_LENGTH,
     DOMAIN_FIELD_MAX_LENGTH,
-    DOMAIN_FIELD_SCALE,
+    DOMAIN_FIELD_SCALE, DOMAIN_FIELD_SCANABLE_OPTION,
```
don't forget to run the linting on this PR: `yarn run lint-branch-fix`

Sigmonia replied:
done

cnathe commented on packages/components/src/internal/components/domainproperties/TextFieldOptions.tsx:
```diff
@@ -75,6 +85,41 @@ export class TextFieldOptions extends React.PureComponent<TextFieldProps, TextFi
+                            disabled={isFieldFullyLocked(lockType)}
+                            checked={scannable}
+                        />
+                        <span>Search this field when scanning samples.</span>
```
```suggestion
                        <span>Search this field when scanning samples</span>
```

Sigmonia replied:
fixed

cnathe commented on packages/components/src/internal/components/domainproperties/models.tsx:
```diff
@@ -1695,6 +1698,7 @@ export interface IDomainFormDisplayOptions {
     hideImportExport?: boolean;
     hideConditionalFormatting?: boolean;
     hideInferFromFile?: boolean;
+    showScannableOption?: boolean;
```
not sure where to put this comment, but I pulled up the PR quick locally and noticed that the new scannable property isn't shown in the grid when you toggle the field editor from Detail mode -> Summary mode.

Sigmonia replied:
Add check for SampleSet and app, and adjusted sort location

Sigmonia committed:
Lint fixes and typo correction

Sigmonia committed:
Remove '.' from text
Adjust location of Scannable column in designer summary
Remove Scannable column for domain summaries where it is N/A

Sigmonia committed:
lint

Sigmonia committed:
merge from develop

Sigmonia committed:
Update tests after CR

Sigmonia committed:
v2.103.0-fb-scanOption.1

Sigmonia committed:
merge 2.103.1 into fb

Sigmonia committed:
Fix scannable option being set for prior to field type change

Sigmonia committed:
merge develop and bump alpha .2

Sigmonia committed:
update snapshot

Sigmonia committed:
Finalize release notes

Sigmonia committed:
v2.105.0
